repository_service_tuf/helpers/tuf.py

```diff
@@ -180,15 +178,21 @@ def _update_snapshot(
             f"signing threshold '{threshold}'"
         )
 
-        root.roles[role_name] = Role([], threshold)
-        for signer in signers:
-            root.add_key(
-                Key.from_securesystemslib_key(signer.key_dict), role_name
-            )
+        roles[role_name] = Role([], threshold)
+        keys: list[tuple[Key, str]] = [
+            (Key.from_securesystemslib_key(signer.key_dict), role_name)
+            for signer in signers
+        ]
```

Comment:
Are you sure the elements of `keys` are `tuples`?
As here on lines 183-184 I see that we create only a `Key` instance.

Reply:
Hm I can see how this can be confusing. If you see `root.add_key` it takes as arguments a `key: Key` and a `role: str`. So, it's one key, with the mapping "key: role" (what I wrote as key[0] which is the key, and key[1] which is the role). I could have used a dict to make it clearer but the Key class would need to be a frozen class in order to use it as a key.

I will change the naming to make it clearer.

Reply:
Thanks, now it's clearer after your explanation and the renaming.